gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegistrationUnit.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.hive;
+
+import java.util.List;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import gobblin.annotation.Alpha;
+import gobblin.configuration.State;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+
+/**
+ * A class that represents a Hive table or partition.
+ *
+ * @author ziliu
+ */
+@Getter
+@Alpha
+public class HiveRegistrationUnit {
+
+  protected final String dbName;
+  protected final String tableName;
+  protected final List<Column> columns = Lists.newArrayList();
+  protected final State props = new State();
+  protected final State storageProps = new State();
+  protected final State serDeProps = new State();
+
+  /**
+   * Table or Partition properties
+   */
+  protected Optional<Long> createTime;
+  protected Optional<Long> lastAccessTime;
```

Comment:
`props` is `Parameters`. 
There are a few properties not included here, such as `Priviledges` and `SortedCols`, since their return types are not String and require adding more classes. If there's a use case that needs them we can add them later